copyright/copyright.js

```diff
@@ -0,0 +1,128 @@
+const defaultBG = '\x1b[49m'
+const defaultFG = '\x1b[39m'
+
+let langs = {}
+let lintMode = true
+let error = false
+
+// we don't want to pass node and copyright.js directories to the glob
+const args = process.argv.filter((arg) => {
+    return !/node|copyright/.test(arg)
+})
+
+/**
+ * getHeaderText - retrieve appropriate header file context from langs{} object
+ * @param  {String} ext file extension of desired header file
+ * @return {String}     content of appropriate header file
+ */
+const getHeaderText = (ext) => {
+    if (!langs[ext]) {
+        console.log(`${red}${blackBG}ERROR${defaultBG} - ${ext} is not supported (yet)`)
+        process.exit(1)
+    } else {
+        return langs[ext]
+    }
+}
+
+/**
+ * buildSupportedExtensions - initializes the langs{} object with the supported
+ * extensions, as well as their respective (c) header content
+ * @return {undefined} - populates the langs{} object
+ */
+const buildSupportedExtensions = () => {
+    const headerDir = path.join(__dirname, './headers')
+    fs
+        .readdirSync(headerDir)
+        .forEach((file) => {
+            const extension = file.match(/\.[0-9a-z]+$/i)[0]
+            const textPath = path.join(headerDir, file)
+            const content = fs.readFileSync(textPath).toString().replace('year', new Date().getFullYear())
+            langs[extension] = content
+        })
+}
+
+
+if (args.length === 0 || args.indexOf('--help') >= 0) {
+
+    console.log(`
+    Usage: node copyright.js [options] 'glob' ['additional globs']
+
+    If your glob is not targetting all nested directories, ensure that the glob string is wrapped in quotes
+
+    Example:
+        ${yellow}node copyright.js --fix${defaultFG} 'src/**/*.js'
+
+        Options:
+
+            --fix        run in fix mode
+
+    Visit ${cyan}https://github.com/mobify/mobify-code-style${defaultFG} to learn more.
+`)
+
+    process.exit(0)
+}
+
+// Sets fix flag if the user provides --fix command line arg
+if (args.indexOf('--fix') >= 0) {
+    args.splice(args.indexOf('--fix'), 1)
+    lintMode = false
+}
+
+buildSupportedExtensions()
+
+args
+    .map((folder) => glob.sync(folder))
+    .reduce((a, b) => a.concat(b)) // flatten array of arrays
+    .forEach((file) => {
+        let content = fs.readFileSync(file)
+
+        const hasCopyrightHeader = content.includes('Copyright (c)')
+        const ext = file.match(/\.[0-9a-z]+$/i)[0]
+
+        if (!hasCopyrightHeader) {
+            if (lintMode) {
+                console.log(`${yellow}${file} ${red}missing copyright header`)
+                error = true
+            } else {
+                const contentStr = content.toString().split('\n')
+                let shebang = ''
+                let newData = ''
+
+                // accomodate for shebang and insert before header
+                if (contentStr[0].indexOf('#!') >= 0) {
+                    shebang = contentStr.shift()
+                    content = contentStr.join('\n')
```

Comment:
I'm not sure if this check for the shebang _feels_ right. It's currently working, but the string manipulation here feels a bit awkward.

<img width="705" alt="screen shot 2017-05-04 at 4 15 12 pm" src="https://cloud.githubusercontent.com/assets/16109837/25728577/e3c39754-30e4-11e7-8c6a-cccc1cd26b5f.png">